### DISCUSSION TITLE
Cannot connect to Pico via USB from Ubuntu

### DISCUSSION THREAD

**Discussion by ctrilsbach** [tabemann/zeptoforth#67] [Q&A] [answered]
I can connect fine to the Pico via USB in Windows using Zeptocom.js but when I try with the same setup in Ubuntu neither Zeptocom.js or e4thcom connects at all. Zeptocom.js reports that that there are no compatible devices found. Which device should it be on ? I am trying /dev/ttyUSB0.

**Comment by tabemann** [comment]
For Linux use /dev/ttyACM0 or like. It would be weird though if that did not show up a device list.

**Reply by ctrilsbach** [reply]
Thanks Travis. Now working in e4thcom with ttyACM0 but still not on Zeptocom.js running on Chrome.

In dmesg it produces the following :-

1: usb 1-10: new full-speed USB device number 10 using xhci_hcd

**Comment by rdoetjes** [comment]
Unplug the deviceThen plug it in and run: sudo dmesg it will show you the device name. I use e4thcom and it works great. And for e4thcom you need to omit the /dev/ just use the ttyUSBx or ttyACMx that shows up in dmesgSent from my iPhoneOn 15 Sep 2023, at 11:13, ctrilsbach ***@***.***> wrote:﻿
I can connect fine to the Pico via USB in Windows using Zeptocom.js but when I try with the same setup in Ubuntu neither Zeptocom.js or e4thcom connects at all. Zeptocom.js reports that that there are no compatible devices found. Which device should it be on ? I am trying /dev/ttyUSB0.

**Reply by ctrilsbach** [reply]
Thanks. ttyACM0 works ok.
 
In dmesg it produces the following :-

1: usb 1-10: new full-speed USB device number 10 using xhci_hcd

**Comment by tabemann** [comment, marked as answer]
I use zeptocom.js on Chrome on Linux as `/dev/ttyACM0` or `/dev/ttyACM1`, depending on what I have connected, and the only catch I have is that if I run anything that uses `codeload3.py` directly or later requires simply connecting to the device with `screen` prior to accepting connections from zeptocom.js.

**Reply by ctrilsbach** [reply]
Strange but not important as e4thcom works ok.
